butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/pom/PomChangePlugin.java

```diff
@@ -0,0 +1,252 @@
+package com.paypal.butterfly.utilities.operations.pom;
+
+import com.paypal.butterfly.extensions.api.TOExecutionResult;
+import com.paypal.butterfly.extensions.api.exception.TransformationOperationException;
+import com.paypal.butterfly.extensions.api.operations.ChangeOrRemoveElement;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.model.PluginExecution;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Changes a Plugin in a Maven POM file.
+ * It allows changing version, extensions, executions and dependencies but not group id and artifact id.
+ * It also allows removing specific configuration, letting them
+ * to have default values, or be managed by applicable.
+ * <br>
+ * If the plugin to be changed doesn't actually exist, it will result
```

Comment:
This should depend actually on the user. That is why this class implements `ChangeOrRemoveElement`. See `IfNotPresent` instance variable. Please change this comment and honor the `ChangeOrRemoveElement` interface.

Reply:
Updated the comment, where it mentions that the default behavior can be changed using `IfNotPresent`